### ISSUE TITLE
`cache-lazy` is an ignored cell option

### ISSUE THREAD

**Issue by prncevince** [quarto-dev/quarto-cli#6491]
### Bug description

`cache-lazy`, as referenced in the [Cache section of the Code Cells: Knitr docs](https://quarto.org/docs/reference/cells/cells-knitr.html#cache), is ignored as a cell option in knitr chunks when using quarto.

This keeps the default knitr option of `cache.lazy=True`.  Setting the chunk option `cache-lazy` does not change this.

```{r}
#| cache: true
#| cache-lazy: false
```

This actually sets `cache.lazy=False` to false

```{r}
#| cache: true
#| cache.lazy: false
```

**Comment by cderv** [comment]
Thanks for the report. 

This is another specific case of 

* https://github.com/quarto-dev/quarto-cli/issues/5852

I'll close this one and focus on the other one. 

Thank you.

**Comment by cderv** [comment]
This has been fixed on **knitr** - from next version 1.44 (and from current dev version 1.43.7), all dash options are changed to their dot version. This includes `cache-lazy`
